### PR TITLE
Synchronize Pitboss time with real Node.js time

### PR DESCRIPTION
**Pull request by honzajavorek**
#### :rocket: Why this change?

Windows tests are extremely flaky. After deeper investigation it turns out [Pitboss](https://www.npmjs.com/package/pitboss-ng) sandbox has different time than the rest of Dredd, so whatever runs within the sandbox can under certain circumstances seem to happen in the future.

#### :memo: Related issues and Pull Requests

- Follow-up to #737 
- Closes https://github.com/apiaryio/dredd/issues/743

#### :white_check_mark: What didn't I forget?

- [x] To write docs
- [x] To write tests
- [x] To put [Conventional Changelog](https://dredd.readthedocs.io/en/latest/contributing/#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

**Comment by honzajavorek** [comment]
@w-vi @miiila Who comes first wins 🏁 This one is tricky, feel free to ask if something isn't clear.

File: src/transaction-runner.coffee [hunk 1]
```diff
@@ -174,6 +175,15 @@ class TransactionRunner
...
     for log in results.logs or []
       @logs.push log
+
+    # For unknown reasons, the sandboxed code in Pitboss runs in "future"
+    # on Windows (and just sometimes - it's flaky). The extreme case is
+    # that sandboxed 'before' hooks can have timestamps with a millisecond
+    # later time then the HTTP transaction itself. Following line
+    # synchronizes the time. It waits until the time of the normal Node.js
+    # runtime happens to be later than time inside the Pitboss sandbox.
+    while Date.now() - results.now < 0 then # ...then do nothing...
```
Comment (w-vi) [inline src/transaction-runner.coffee]: Dirty hack but should work, would not a sleep as in `setTimeout` work better. Depends on how out of sync the time is though.

Comment (honzajavorek) [inline src/transaction-runner.coffee]: It's one millisecond. I have the same concerns, but wanted your opinion on whether you think one millisecond is too much to have active waiting. You're the as close to HW as possible person here 😉

Comment (honzajavorek) [inline src/transaction-runner.coffee]: Discussed in person - 1 millisecond is OK (too small time)

**Pull request merged**